### ISSUE TITLE
Invalid OS_VARIANT setting on VM creation

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#177]
Originally reported of Google Code with ID 121.

```
(in 2.2 branch)
OS_VARIANT is always set to first variant in variants.list This is due to small bug introduced in efaa9b06d1e1e6d1678d0edd75b1ba37cf0de3d9

in backend.py, OSCoreEnv():
  inst_os.name is pure operating system name (without variant) as variant is stripped in OSFromDisk(). So we always get variant = inst_os.supported_variants[0] (first variant in variants list).
  My small patch (attached) adds argument os_name with full name (including variant) to solve this problem.

```
Originally added on 2010-09-08 17:13:59 +0000 UTC.

Attachments: 
[0001-Fix-OS_VARIANT-variable-setting-introduced-in-efaa9b.patch](https://storage.googleapis.com/google-code-attachments/ganeti/issue-121/comment-0/0001-Fix-OS_VARIANT-variable-setting-introduced-in-efaa9b.patch)

**Comment by Ganeti-Issues-Migrator** [comment]
```
I confirm this patch fixes this issue.
```
Originally added on 2010-09-09 02:48:49 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This looks good, but you missed one other call to OSCoreEnv. I'll apply a modified version.

thank!
iustin
```
Originally added on 2010-09-13 09:49:00 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed on the 2.2 stable branch, will be included in 2.2.0.

thanks again,
iustin
```
Originally added on 2010-09-13 11:07:03 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.2

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-10-22 08:57:08 +0000 UTC.

Changed State: Released
